src/evaluate_location.py

```diff
@@ -29,27 +29,47 @@ def evaluate_location(location_data: dict) -> dict:
         location_data
     )
 
-    loc_feas = loc_res[loc_res['Feasible'] == 1].copy()
+    loc_feas = []
+    for key, value in loc_res.items():
+        if value['Feasible'] == 1:
+            loc_feas.append(value)
 
-    if not loc_feas.empty:
+    if len(loc_feas) > 0:
+        machines_value = []
+        for result in loc_feas:
+            machines_value.append(result['Machines'])
+        
         # find fewest feasible machines
-        mach_min = loc_feas['Machines'].min()
+        mach_min = min(machines_value)
 
         # keep the feasible setup with the fewest number of machines
-        loc_feas_min = loc_feas[loc_feas['Machines'] == mach_min].copy()
+        loc_feas_min = {}
+        for key, value in loc_res.items():
+            if value['Machines'] == mach_min:
+                loc_feas_min = value
+                break
 
         # populate overall results with info for this location
+        
         best_result['Resource'] = mach_min
-        best_result['Exp. Avg. Wait Time'] = loc_feas_min.iloc[0]['BatchAvg']
-        best_result['Exp. Max. Wait Time'] = loc_feas_min.iloc[0]['BatchMaxAvg']
+        best_result['Exp. Avg. Wait Time'] = loc_feas_min['BatchAvg']
+        best_result['Exp. Max. Wait Time'] = loc_feas_min['BatchMaxAvg']
+        
     else:
         # no feasible setups, find lowest wait time (should work out to be max machines allowed)
-        wait_time_min = loc_res['BatchMaxAvg'].min()
+        max_avg = []
+        for key, result in loc_res.items():
+            max_avg.append(result['BatchMaxAvg'])
+        wait_time_min = min(max_avg)
 
-        loc_res_min = loc_res[loc_res['BatchMaxAvg'] == wait_time_min]
+        loc_res_min = {}
+        for key, result in loc_res.items():
+            if result['BatchMaxAvg'] == wait_time_min:
+                loc_res_min = result
+                break
```

Comment:
Theres no need to loop twice here anymore. Since this is a dict now, just keep track of the dict index that had the min wait time.